### ISSUE TITLE
[BUG] Pytorch dataloader bug; invalid pointer error

### ISSUE THREAD

**Issue by EvenOldridge** [NVIDIA/NVTabular#320]
**Describe the bug**
Pytorch dataloader is hitting an invalid pointer.  "Invalid __shared__ write of size 8 bytes"

Likely a thrust issue, and possibly an issue on the cuDF side

**Steps/Code to reproduce bug**
python 3.7 with rapids and nvtabular

The error is reproducible using this code snippet:
import torch
import glob
import os
import nvtabular as nvt
from nvtabular.ops import Normalize,  Categorify,  LogOp, FillMissing, Clip, get_embedding_sizes
from nvtabular.loader.torch import TorchAsyncItr, DLDataLoader

output_train_dir = '/tmp/pytest-of-root/pytest-0/test_criteo_notebook0' # containing the day_x.parquet files
PARTS_PER_CHUNK = 2
BATCH_SIZE = 800000
CATEGORICAL_COLUMNS = ['C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'C10', 'C11', 'C12', 'C13', 'C14', 'C15', 'C16', 'C17', 'C18', 'C19', 'C20', 'C21', 'C22', 'C23', 'C24', 'C25', 'C26']
CONTIGUOUS_COLUMNS = ['I1', 'I2', 'I3', 'I4', 'I5', 'I6', 'I7', 'I8', 'I9', 'I10', 'I11', 'I12', 'I13']
LABEL_COLUMNS = ['label']

train_paths = glob.glob(os.path.join(output_train_dir, "*.parquet"))
train_data = nvt.Dataset(train_paths, engine="parquet", part_mem_fraction=0.04/PARTS_PER_CHUNK)
train_data_itrs = TorchAsyncItr(
    train_data,
    batch_size=BATCH_SIZE,
    cats=CATEGORICAL_COLUMNS,
    conts=CONTIGUOUS_COLUMNS,
    labels=LABEL_COLUMNS,
    parts_per_chunk=PARTS_PER_CHUNK
)

torch.cuda.synchronize()
print(train_data_itrs)
print(next(train_data_itrs))
I assume the thrust/cub method is called in `nvt.Dataset` or the `TorchAsyncItr`. Unfortunately, I wasn't able to isolate the it further, as I'm not familiar enough with the implementations of these two nvtabular classes.

`compute-sanitizer --tool memcheck` yields:
========= Invalid __shared__ write of size 8 bytes
=========     at 0x310 in void thrust::cuda_cub::core::_kernel_agent<thrust::cuda_cub::__copy_if::CopyIfAgent<thrust::counting_iterator<long,thrust::use_default,thrust::use_default,thrust::use_default>,thrust::device_ptr<bool>,thrust::permutation_iterator<thrust::device_ptr<long>,thrust::transform_iterator<strided_range<thrust::device_ptr<long>>::stride_functor,thrust::counting_iterator<long,thrust::use_default,thrust::use_default,thrust::use_default>,thrust::use_default,thrust::use_default>>,NonZeroOp<bool>,int,int*>,thrust::counting_iterator<long,thrust::use_default,thrust::use_default,thrust::use_default>,thrust::device_ptr<bool>,thrust::permutation_iterator<thrust::device_ptr<long>,thrust::transform_iterator<strided_range<thrust::device_ptr<long>>::stride_functor,thrust::counting_iterator<long,thrust::use_default,thrust::use_default,thrust::use_default>,thrust::use_default,thrust::use_default>>,NonZeroOp<bool>,int,int*,cub::ScanTileState<int,bool=1>,unsigned long>(thrust::use_default,thrust::use_default,thrust::use_default,thrust::counting_iterator<long,thrust::use_default,thrust::use_default,thrust::use_default>,bool,thrust::device_ptr<bool>,long,thrust::device_ptr<long>)
=========     by thread (96,0,0) in block (0,0,0)
=========     Address 0xc00 is out of bounds
=========     Saved host backtrace up to driver entry point at kernel launch time
=========     Host Frame:cuLaunchKernel [0x7f63356de25e]
=========                in /usr/lib/x86_64-linux-gnu/libcuda.so.1
=========     Host Frame: [0x7f631fce262b]
=========                in /usr/local/cuda/lib64/libcudart.so.11.0
=========     Host Frame:cudaLaunchKernel [0x7f631fd225b1]
=========                in /usr/local/cuda/lib64/libcudart.so.11.0
=========     Host Frame:long thrust::cuda_cub::copy_if<thrust::detail::execute_with_allocator<THCThrustAllocator&, thrust::cuda_cub::execute_on_stream_base>, thrust::counting_iterator<long, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::device_ptr<bool>, thrust::permutation_iterator<thrust::device_ptr<long>, thrust::transform_iterator<strided_range<thrust::device_ptr<long>>::stride_functor, thrust::counting_iterator<long, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::use_default, thrust::use_default>>, NonZeroOp<bool>>(thrust::cuda_cub::execution_policy<THCThrustAllocator&>&, thrust::cuda_cub::execute_on_stream_base, thrust::cuda_cub::execution_policy<THCThrustAllocator&>&, thrust::detail::execute_with_allocator<THCThrustAllocator&, thrust::cuda_cub::execute_on_stream_base>, thrust::cuda_cub::copy_if<thrust::detail::execute_with_allocator<THCThrustAllocator&, thrust::cuda_cub::execute_on_stream_base>, thrust::counting_iterator<long, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::device_ptr<bool>, thrust::permutation_iterator<thrust::device_ptr<long>, thrust::transform_iterator<strided_range<thrust::device_ptr<long>>::stride_functor, thrust::counting_iterator<long, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::use_default, thrust::use_default>>, NonZeroOp<bool>>, thrust::use_default) [0x7f62e697f535]
=========                in /opt/conda/envs/rapids/lib/python3.7/site-packages/torch/lib/libtorch_cuda.so
=========     Host Frame:THCudaBoolTensor_nonzero(THCState*, c10::TensorImpl*, c10::TensorImpl*) [0x7f62e6973111]
=========                in /opt/conda/envs/rapids/lib/python3.7/site-packages/torch/lib/libtorch_cuda.so
[...]
Attaching the `compute-sanitizer` output.

**Expected behavior**

**Environment details (please complete the following information):**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of NVTabular install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
 

**Additional context**

**Comment by benfred** [comment]
See also https://github.com/rapidsai/cudf/issues/6295

**Comment by jperez999** [comment]
Still verify but recent tests show this is not a problem in 0.16

**Comment by jperez999** [comment]
verified nightly does not show this behavior anymore.

**Comment by EvenOldridge** [comment]
What should we do about the containers?
